SofaKernel/modules/SofaCore/src/sofa/core/ObjectFactory.h

```diff
@@ -72,7 +72,8 @@ class SOFA_CORE_API ObjectFactory
         /// BaseClass structure associated with the type of intanciated objects.
         virtual const objectmodel::BaseClass* getClass() = 0;
 
-        virtual std::string shortName(objectmodel::BaseObjectDescription* arg) = 0;
+        SOFA_ATTRIBUTE_DISABLED__CLASSNAME_INTROSPECTION()
+        std::string shortName(objectmodel::BaseObjectDescription* arg) = delete;
```

Comment:
Not 100% sure about the changes in this file :thinking: 
@damienmarchal Could you validate?